From e57066e2615d859b3d65cefb9f41f0907210fad3 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 13 Feb 2020 01:51:22 +0200
Subject: [PATCH] [SW-8229] goya2: fix writel switch offset bug

Some writel where switched between offset and reg value
to write. First parameter should be reg and then offset
but on some writel operations this was switched around.

Change-Id: I1741cf009bd4292a5cee638681d29a4a2fa28a41
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 208ae5a..8d3ae29 100644
--- a/common/security.c
+++ b/common/security.c
@@ -114,14 +114,14 @@
 
 void security_spi_dma_mmu_bypass(void)
 {
-	regmap_writel(mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER,
-		      MSTR_IF_AXUSER_HB_SEC_MMBP);
+	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
+		      mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER);
 }
 
 void security_mem_cpy_mmu_bypass(void)
 {
-	regmap_writel(mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER,
-		      MSTR_IF_AXUSER_HB_SEC_MMBP);
+	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
+		      mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER);
 }
 
 static void secuirty_elbi_rr_lbw_config(uint8_t spi_controller_en)
diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
index a0b1b12..ff82160 100644
--- a/goya2_arc/asic.c
+++ b/goya2_arc/asic.c
@@ -15,5 +15,5 @@
 	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
 	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
 	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
-	regmap_writel(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, reg);
+	regmap_writel(reg, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
 }
diff --git a/goya2_arc/pldm.c b/goya2_arc/pldm.c
index a0b1b12..f8063d0 100644
--- a/goya2_arc/pldm.c
+++ b/goya2_arc/pldm.c
@@ -11,9 +11,12 @@
 {
 	uint32_t reg;
 
+	/* disable global masking of prstn reset */
+	regmap_writel(0, mmPSOC_GLOBAL_CONF_PRSTN_MASK);
+
 	/* on warm reset change default to reset ARC and PSOC */
 	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
 	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
 	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
-	regmap_writel(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, reg);
+	regmap_writel(reg, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
 }
